lib/mix/lib/mix/dep.ex

```diff
@@ -31,6 +31,8 @@ defmodule Mix.Dep do
       the information on this field is private to the manager and should not be
       relied on
 
+    * `env_vars` - an enumerable of tuples containing environment key-value as binary
```

Comment:
This needs to be elaborated on. It doesn't say what the option does and when it's used. For example it is only used during loading and not during compilation for example.

Reply:
I copied the text of the `:env` option of [System.cmd/3](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/system.ex#L566) which is of the same type. You're absolutely right that it needs better doc there, I'll update it.

Should I also add an entry in the changelog about this?

Reply:
We try to avoid updating the changelog in PRs because it easily leads to conflicts. An entry will be added before we do a new release of course.

Reply:
Reworded it to:
> an enumerable of key-value tuples of binaries to be set as environment variables when loading or compiling the dependency

I tried to keep keep it short to match the terseness of the rest of the options. Let me know what you think.